homeassistant/components/shelly/config_flow.py

```diff
@@ -460,7 +483,7 @@ def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         return (
             get_device_entry_gen(config_entry) in RPC_GENERATIONS
             and not config_entry.data.get(CONF_SLEEP_PERIOD)
-            and config_entry.data.get("model") != MODEL_WALL_DISPLAY
+            and config_entry.data.get(CONF_SCRIPT, True)
```

Comment:
We need the `config_entry.data.get("model") != MODEL_WALL_DISPLAY` condition here in addition, the update of the entry data will happen after this is evaluated and will be incorrect